Topics/Tech_Stacks/Angular Basics.md

```diff
@@ -0,0 +1,182 @@
+
+# Understanding Angular: A Guide for Beginners
+
+## 1. What are Components?
+
+### Definition
+In Angular, components are the fundamental building blocks. They control a part of the UI for your application. Each component consists of three main parts:
+
+- **Template**: This is the HTML part, defining how the component looks.
+- **Class**: Written in TypeScript, it handles the data and behavior of the component.
+- **Decorator**: This provides metadata about the component, like its selector and template URL.
+
+For example
+
+**list.component.ts** (files are stored as .ts, guess what ts stands for - hint: it's not Top Secret)
+```
+import { Component } from '@angular/core';
+
+@Component({
+  selector: 'app-list',
+  template: `list.component.html`,
+})
+export class ListComponent {
+    users = {
+        { name: 'John', age: 20 },
+        { name: 'Jane', age: 21 },
+        { name: 'Jack', age: 22 },
+    }
+}
+```
+
+**list.component.html**
+```
+<ul>
+    <li *ngFor="let user of users">
+        {{ user.name }} is {{ user.age }} years old.
+    </li>
+</ul>
+```
+
+I think the easiest way to look at a Component is a Class which will be shown on your screen. It may be shown multiple times, and may require some different data to be shown. For example, a list of users may be shown on the screen, and each user may have a different name, age, and other information. The component is the class which will be used to show the user on the screen, and the data will be passed in to the component.
+
+
+### Usage
+
+You created a list component and some other cool looking components for your page. Now, how do you use it?
+
+Not following angular conventions, you just put them in a mega html called
+
+**app.component.html**
+```
+<app-list></app-list>
+<app-cool-component></app-cool-component>
+```
+
+## 2. Component Interaction
+
+### Inheritance
+Components can inherit from other components, sharing functionalities like methods or properties. Think of it like a family tree where children inherit traits from their parents.
+
+### Communication
+- **Input**: Components can receive data from their parent using `@Input()`. It's like getting a present; the parent component sends it, and the child receives it.
+- **Output**: To communicate back to the parent, a child component uses `@Output()` and EventEmitter. It's like the child sending a letter back home.
+
+Imagine you wanted to dynamically input users into your list component and output the oldest user. You could do something like this:
+
+**app.component.html**
+```
+<app-list [users]="usersFromAppComponent" (oldestUser)="oldestUser($event)"></app-list>
```

Comment:
It might make it clearer if you explain what $event does/is

Reply:
Yes, that'd be helpful. Added the explaination